keep/providers/appdynamics_provider/appdynamics_provider.py

```diff
@@ -0,0 +1,280 @@
+                self.authentication_config.appDynamicsPassword)
+
+    def __create_http_response_template(self, keep_api_url: str):
+        keep_api_host, keep_api_path = keep_api_url.rsplit("/", 1)
+
+        temp = tempfile.NamedTemporaryFile(mode='w+t', delete=True)
+
+        template = json.load(open(r'./httpactiontemplate.json'))
+        template["host"] = keep_api_host
+        template["path"] = keep_api_path
+
+        temp.write(json.dumps(template))
+        temp.seek(0)
+
+        res = requests.post(self.__get_url(paths=["controller/actiontemplate/httprequest"]),
+                            files={"template": temp}, auth=self.__get_auth())
+        res = res.json()
+        temp.close()
+        if res["success"] == "True":
+            logger.info("HTTP Response template Successfully Created")
+        else:
+            logger.info("HTTP Response template creation failed", extra=res)
+            if 'already exists' in res['error'][0]:
+                logger.info("HTTP Response template creation failed as it already exists", extra=res)
+                raise HTTPActionTemplateAlreadyExists()
+            raise Exception(res["errors"])
+
+    def setup_webhook(
+            self, tenant_id: str, keep_api_url: str, api_key: str, setup_alerts: bool = True
+    ):
+        try:
+            self.__create_http_response_template(keep_api_url=keep_api_url)
+        except HTTPActionTemplateAlreadyExists:
+            logger.info("Template already exists, proceeding with webhook setup")
+        except Exception as e:
+            raise e
+
+        # Listing all policies in the specified app
+        policies_response = requests.get(
+            url=self.__get_url(paths=["alerting/rest/v1/applications", self.authentication_config.appName, "policies"]),
+            auth=self.__get_auth(),
+        )
+
+        policies = policies_response.json()
+        policy_config = {
+            "actionName": "KeepWebhook",
+            "actionType": "HTTP_REQUEST",
+        }
+        for policy in policies:
+            if policy_config not in policy["actions"]:
+                policy["actions"].append(policy_config)
+            request = requests.put(
+                url=self.__get_url(
+                    paths=["/alerting/rest/v1/applications", self.authentication_config.appName, "policies",
+                           policy["id"]]),
+                auth=self.__get_auth(),
+                data=policy,
+            )
+            if not request.ok:
+                logger.error("Failed to add Webhook", extra=request.json())
+                raise Exception("Could not create webhook")
+            logger.info("Webhook created")
+
+    @staticmethod
+    def _format_alert(
+            event: dict,
+            provider_instance: Optional["AppdynamicsProvider"],
+    ) -> AlertDto:
+        return AlertDto(
+            **event,
+            severity=AppdynamicsProvider.SEVERITIES_MAP.get(event['severity']),
```

Comment:
Did you test this? it'll fail because you pass severity twice (**event and severity=)